homeassistant/components/utility_meter/sensor.py

```diff
@@ -396,7 +439,18 @@ async def async_added_to_hass(self):
             )
         )
 
-        if state := await self.async_get_last_state():
+        if (last_sensor_data := await self.async_get_last_sensor_data()) is not None:
+            # new introduced in 2022.04
+            self._state = last_sensor_data.native_value
+            self._unit_of_measurement = last_sensor_data.native_unit_of_measurement
+            self._last_period = last_sensor_data.last_period
+            self._last_reset = last_sensor_data.last_reset
+            if last_sensor_data.status == COLLECTING:
+                # Fake cancellation function to init the meter in similar state
```

Comment:
There's nothing fake about this, the null lambda is perfectly fine 👍 
```suggestion
                # Null lambda to allow cancelling the collection on tariff change
```